plugin/vcs/gitlab/gitlab.go

```diff
@@ -288,20 +288,27 @@ func (provider *Provider) FetchRepositoryActiveMemberList(ctx context.Context, o
 	}
 
 	// we only return active member (both state and membership_state is active)
+	var emptyEmailUserIDList []string
 	var activeRepositoryMemberList []*vcs.RepositoryMember
 	for _, gitLabMember := range gitLabrepositoryMember {
 		if gitLabMember.State == vcs.StateActive {
-			// the email field does not return as expected via projects/<<projectId>>/members/all, possibly caused by: https://gitlab.com/gitlab-org/gitlab/-/issues/25077
-			// so we need to fetch this field one by one
+			// the email field does not return if the user does not have the admin accessibility
+			// for normal user, email field will not be returned by GitLab
+			// thus we use public email
+			// TODO: need to work around if the user does not set public email (for now we just return an error showing user who do not configure their public emails)
```

Comment:
```suggestion
			// The email field will only be returned if the caller credential is associated with a GitLab admin account.
			// And since most callers are not GitLab admins, thus we fetch public email
			// TODO: need to work around this if the user does not set public email. For now, we just return an error listing users not having public emails.
```